### PR TITLE
feat: add write amplitude model from STM

### LINKED ISSUE #91: HelicityAmplitudeGenerator should warn if not using the correct formalism
Currently, it is possible to use the expert system with, say, the `StateTransitionManager` with argument `formalism_type='canonical-helicity'` in combination with `HelicityAmplitudeGeneratorXML` and vice versa. An exception should be raised in this situation, because the combination physically doesn't make sense.

### PR DESCRIPTION
**Pull request by redeboer**
Closes #91 

Amplitude model recipe writing can now be done directly from the STM. This removes the need to first create e.g. a `HelicityAmplitudeGenerator`.

### CODE REVIEW

File: expertsystem/ui/__init__.py [hunk 1]
```diff
@@ -239,84 +339,6 @@ def _determine_node_settings(self, graphs):
             graph_node_setting_pairs.append((instance, node_settings))
         return graph_node_setting_pairs
 
-    def find_solutions(
```
Comment (spflueger) [inline expertsystem/ui/__init__.py:242]: Why did you move this up?  My logic here was to keep functions called by other functions close to each other, so that the user does not need to scroll around too much

Comment (redeboer) [reply expertsystem/ui/__init__.py:242]: To keep the public functions together (1c7216aec94b0370b8f58ae7113fe455debc6394). It's more for the developer, in that sense it's indeed better to keep it on the same place

File: expertsystem/ui/__init__.py [hunk 2]
```diff
@@ -72,6 +74,7 @@ def __init__(  # pylint: disable=too-many-arguments
                 f"Formalism type {formalism_type} not implemented."
                 f" Use {allowed_formalism_types} instead."
             )
+        self.__formalism_type = formalism_type
```
Comment (spflueger) [inline expertsystem/ui/__init__.py:77]: Actually a lot of the other attributes should also be made into private variables or properties etc

Comment (redeboer) [reply expertsystem/ui/__init__.py:77]: Yep... working on that

Comment (redeboer) [reply expertsystem/ui/__init__.py:77]: but better to do that in a separate PR

File: expertsystem/ui/__init__.py (hunk 1 above)
Comment (redeboer) [reply expertsystem/ui/__init__.py]: 18e8684d48fcb993e71451578fd688a335873009

File: expertsystem/ui/__init__.py (hunk 2 above)
Comment (spflueger) [reply expertsystem/ui/__init__.py:77]: ok

File: expertsystem/ui/__init__.py (hunk 1 above)
Comment (spflueger) [reply expertsystem/ui/__init__.py]: Ok. Although I would be fine with such a change in general. We should have some kind of decision on this and then write this down in and adr or the docs. I want to avoid such changes going back and forth and having no consistent manner.

Comment (redeboer) [reply expertsystem/ui/__init__.py]: It's hard to pin down. Sometimes you need to have the class defined first. Maybe one prefers public methods first. And then in some cases having the definition coming right after the call is easier for readability.

Comment (spflueger) [reply expertsystem/ui/__init__.py]: I just followed the recommendation from clean code i believe (which made sense to me). I guess nowadays with strong IDEs this not as crucial anymore, but i think its still kind of nice to have thinks stuck together. So I would actually vote for how it is now. But I can be convinced otherwise :)

Comment (redeboer) [reply expertsystem/ui/__init__.py]: Yep it's from Clean Code ^^
